easybuild/easyblocks/f/flexiblas.py

```diff
@@ -0,0 +1,137 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing FlexiBLAS, implemented as an easyblock
+
+author: Kenneth Hoste (HPC-UGent)
+"""
+import os
+
+from easybuild.easyblocks.generic.cmakemake import CMakeMake
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+
+class EB_FlexiBLAS(CMakeMake):
+    """Support for building/installing FlexiBLAS."""
+
+    @staticmethod
+    def extra_options():
+        extra_vars = CMakeMake.extra_options()
+        extra_vars.update({
+            'blas_auto_detect': [False, "Let FlexiBLAS autodetect the BLAS libraries during configuration", CUSTOM],
+            'enable_lapack': [True, "Enable LAPACK support, also includes the wrappers around LAPACK", CUSTOM],
+            'flexiblas_default': [None, "Default BLAS lib to set at compile time. If not defined, " +
+                                  "the first BLAS lib in the list of dependencies is set as default", CUSTOM],
+        })
+        extra_vars['separate_build_dir'][0] = True
+        return extra_vars
+
+    def __init__(self, *args, **kwargs):
+        """Easyblock constructor."""
+        super(EB_FlexiBLAS, self).__init__(*args, **kwargs)
+
+        build_dep_names = set(dep['name'] for dep in self.cfg.dependencies(build_only=True))
+        dep_names = [dep['name'] for dep in self.cfg.dependencies()]
+        self.blas_libs = [x for x in dep_names if x not in build_dep_names]
+
+    def configure_step(self):
+        """Custom configuration for FlexiBLAS, based on which BLAS libraries are included as dependencies."""
+
+        configopts = {}
+
+        if self.cfg['blas_auto_detect'] is True:
+            configopts.update({'BLAS_AUTO_DETECT': 'ON'})
+        else:
+            configopts.update({'BLAS_AUTO_DETECT': 'OFF'})
+
+        if self.cfg['enable_lapack'] is True:
+            configopts.update({'LAPACK': 'ON'})
+        else:
+            configopts.update({'LAPACK': 'OFF'})
+
+        supported_blas_libs = ['OpenBLAS', 'BLIS', 'NETLIB']
+        flexiblas_default = self.cfg['flexiblas_default']
+        if flexiblas_default is None:
+            flexiblas_default = self.blas_libs[0]
+
+        if flexiblas_default not in supported_blas_libs:
+            raise EasyBuildError("%s not in list of supported BLAS libs %s", flexiblas_default, supported_blas_libs)
+
+        configopts.update({'FLEXIBLAS_DEFAULT': flexiblas_default})
+
+        # list of BLAS libraries to use is specified via -DEXTRA=...
+        configopts['EXTRA'] = ';'.join(self.blas_libs)
```

Comment:
Maybe check self.blas_libs against supported_blas_libs? E.g. `if any(lib not in supported_blas_libs for lib in self.blas_libs)` or similar

Reply:
I'm not sure we should be doing that `supported_blas_libs` check at all...

@smoors What was your intention with that exactly? Make sure that people don't try to use the `FlexiBLAS` easyblock with `imkl` until we've figured it out?

Reply:
@boegel not only that. also:
-if `flexiblas_default` is **not** set, as a check if the first dependency is actually a (supported) BLAS lib
-if `flexiblas_default` is set, as a check if user makes a syntax error in the BLAS lib

Reply:
Wouldn't `FlexiBLAS` itself complain in that case though?

Reply:
sure, but isn't it nicer if EB exists early with a clear message instead of letting the user search through the logs for the error?

Reply:
Seems like that's a no. ` -DFLEXIBLAS_DEFAULT='foo'` doesn't result in any failures... :(

Reply:
> sure, but isn't it nicer if EB exits early with a clear message instead of letting the user search through the logs for the error?

It is, but we should also try to avoid putting a maintenance burden on ourselves.
Although I guess new BLAS/LAPACK libraries won't exactly be popping up as mushrooms in the near future...